### ISSUE TITLE
RunWorkerAsync runs twice and causes an InvalidOperationException

### ISSUE THREAD

**Issue by mjcortejo** [rnand/Youtube-downloader#11]
The issue is similar to my first commit, whenever I'm changing the url of the video while the program is retrieving the title, it fires off an InvalidOperationException.

I believe it was caused by another RunWorkerAsync() when the condition hits the else statement in:

```
        if (!YTtitlebackgroundWorker.IsBusy)
        {
            YTtitlebackgroundWorker.RunWorkerAsync(urlData);
        }
        else
        {
            YTtitlebackgroundWorker.CancelAsync();
            YTtitlebackgroundWorker.RunWorkerAsync(urlData);
        }
```

Which causes to run another runworkerasync if the backgroundworker is busy.
I have tried a small but temporary fix by removing the entire block of the else statement and simply replaced it with a messagebox saying that the program is currently retrieving the title:

```
        else
        {
            MessageBox.Show("The program is already retrieving the title");
            //YTtitlebackgroundWorker.CancelAsync();
            //YTtitlebackgroundWorker.RunWorkerAsync(urlData);
        }
```

I can run a commit to this, but I have to test it out if it affects the downloading process.

**Comment by mjcortejo** [comment]
Looks like it wasn't affecting the retrieval of the title when the messagebox pops up.

**Comment by rnand** [comment]
Is this fixed now or does this exception still occur? I tried to check it but title retrieve is too fast for me to change the url while it is retrieving. Do you have a slow internet connection?

**Comment by mjcortejo** [comment]
Yes, I tested this on a slow connection. I fixed this on commit f48f1da6fee811a9f28e2e0a81d31531ecf1b2c4. I believe the exception doesn't allow more than one .RunWorkerAsync() if the worker is already busy. The title is still being retrieved even when you change URLs fast.

**Comment by rnand** [comment]
Alright, then I'm closing this issue.
